pkg/cli/cmds/k3sopts.go

```diff
@@ -94,3 +83,61 @@ func commandFromK3S(cmd *cli.Command, flagOpts map[string]*K3SFlagOption) (*cli.
 	cmd.Flags = newFlags
 	return cmd, errs.Err()
 }
+
+// flagSetUsage receives a flag and a K3S flag option, parses
+// both and sets the necessary fields based on the underlying
+// flag type.
+func flagSetUsage(flag cli.Flag, opt *K3SFlagOption) {
+	v := reflect.ValueOf(flag).Elem()
+	if v.CanSet() {
+		switch t := flag.(type) {
+		case *cli.StringFlag:
+			t.Usage = opt.Usage
+		case *cli.StringSliceFlag:
+			t.Usage = opt.Usage
+		case *cli.BoolFlag:
+			t.Usage = opt.Usage
+		}
+	}
+}
+
+// flagSetDefault receives a flag and a K3S flag option, parses
+// both and sets the necessary fields based on the underlying
+// flag type.
+func flagSetDefault(flag cli.Flag, opt *K3SFlagOption) {
+	v := reflect.ValueOf(flag).Elem()
+	if v.CanSet() {
+		switch t := flag.(type) {
+		case *cli.StringFlag:
+			x := opt.Default
+			t.DefaultText = x
+			t.Destination = &x
+			t.Value = opt.Default
+		case *cli.StringSliceFlag:
+			t.DefaultText = opt.Default
+			t.Destination = &([]string{opt.Default})
+			t.Value = []string{opt.Default}
+		case *cli.BoolFlag:
+			t.DefaultText = opt.Default
+			t.Destination = &opt.Hide
+			t.Value = opt.Hide
+		}
+	}
+}
+
+// flagSetHide receives a flag and a K3S flag option, parses
+// both and sets the necessary fields based on the underlying
+// flag type.
+func flagSetHide(flag cli.Flag, opt *K3SFlagOption) {
+	v := reflect.ValueOf(flag).Elem()
+	if v.CanSet() {
+		switch t := flag.(type) {
```

Comment:
As with Usage - if all types support Hidden can we skip the type check?

Reply:
See comment in usage.